### PR TITLE
Okapi login and authorized requests

### PR DESCRIPTION
**Pull request by wwelling**

### CODE REVIEW

**Review by kaladay — CHANGES_REQUESTED** [review]

File: service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java [hunk 1]
```diff
@@ -399,4 +445,65 @@ private Date folioDateToDate(String folioDate) throws ParseException {
...
+
+    private String getToken() {
+        Optional<String> token = TokenUtility.getToken(getName());
+        if (token.isPresent()) {
+            return token.get();
+        }
+        return login();
+    }
+
+    private String login() {
```
[the remark this answers is not in the document]
Comment (wwelling) [reply service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java:480]: Better to be explicit. Will update.

File: service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java [hunk 2]
```diff
@@ -399,4 +445,65 @@ private Date folioDateToDate(String folioDate) throws ParseException {
...
+        String url = properties.getBaseOkapiUrl() + "/authn/login";
+        HttpEntity<Credentials> entity = new HttpEntity<>(properties.getCredentials(), headers(properties.getTenant()));
+        ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.POST, entity, String.class);
```
[the remark this answers is not in the document]
Comment (wwelling) [reply service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java:483]: This does not afford typed generic ResponseEntity.

[the remark this answers is not in the document]
Comment (wwelling) [reply service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java:484]: image

[the remark this answers is not in the document]
Comment (wwelling) [reply service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java:484]: Will use `RepsonseEntity<?>` as suggested in slack.

**Review by kaladay — APPROVED** [review]

**Pull request merged**